src/pipecat/services/resembleai/tts.py

```diff
@@ -0,0 +1,286 @@
+            
+            if len(self._current_audio_buffer) > 8192:  # 8KB chunks
+                await self._send_audio_chunk()
+        else:
+            try:
+                msg = json.loads(message)
+                if msg.get("type") == "audio_end":
+                    if self._current_audio_buffer:
+                        await self._send_audio_chunk()
+                    await self._handle_audio_end(msg)
+                elif msg.get("type") == "error":
+                    await self._handle_error(msg)
+            except json.JSONDecodeError:
+                # If it's not JSON, treat as binary audio data
+                if isinstance(message, str):
+                    message = message.encode()
+                self._current_audio_buffer.extend(message)
+
+    async def _send_audio_chunk(self):
+        if self._current_audio_buffer:
+            frame = TTSAudioRawFrame(
+                audio=bytes(self._current_audio_buffer),
+                sample_rate=self.sample_rate,
+                num_channels=1,
+            )
+            await self.push_frame(frame)
+            self._current_audio_buffer = bytearray()
+
+    async def _handle_json_message(self, message: str):
+        msg = json.loads(message)
+        request_id = msg.get("request_id")
+        
+        if not request_id and not self._request_id and request_id != self._request_id:
+            return
+        
+        if msg["type"] == "audio_end":
+            await self._handle_audio_end(msg)
+            
+        elif msg["type"] == "audio":
+            await self.stop_ttfb_metrics()
+            self.start_word_timestamps()
+            
+            audio_data = base64.b64decode(msg["audio_content"])
+            
+            frame = TTSAudioRawFrame(
+                audio=audio_data,
+                sample_rate=msg.get("sample_rate", self.sample_rate),
+                num_channels=1,
+            )
+            await self.push_frame(frame)
+            
+        elif msg["type"] == "error":
+            await self._handle_error(msg)
+
+    async def _handle_audio_end(self, msg: dict):
+        await self.stop_ttfb_metrics()
+        await self.add_word_timestamps([("TTSStoppedFrame", 0), ("Reset", 0)])
+        await self.push_frame(TTSStoppedFrame())
+        self._is_receiving_audio = False
+
+    async def _handle_error(self, msg: dict):
+        error_message = msg.get('message', 'Unknown error')
+        logger.error(f"{self} error: {error_message}")
+        await self.push_frame(TTSStoppedFrame())
+        await self.stop_all_metrics()
+        await self.push_error(ErrorFrame(f"{self} error: {error_message}"))
+        self._is_receiving_audio = False
+        self._current_audio_buffer = bytearray()
+
+    async def run_tts(self, text: str) -> AsyncGenerator[Frame, None]:
+        logger.debug(f"{self}: Generating TTS [{text}]")
+        
+        try:
+            if not self._websocket or self._websocket.closed:
```

Comment:
Support for the new websocket asyncio implementation.

```suggestion
            if not self._websocket or self._websocket.state is State.CLOSED:
```